aztec/src/main/kotlin/org/wordpress/aztec/source/Format.kt

```diff
@@ -6,22 +6,27 @@ import java.util.regex.Pattern
 object Format {
 
     // list of block elements
-    private val block = "div|span|br|blockquote|ul|ol"
+    private val block = "div|span|br|blockquote|ul|ol|li|h1|h2|h3|h4|h5|h6"
 
     fun addFormatting(content: String): String {
-        // let's use Jsoup for HTML formatting for now because it works out of the box
         val doc = Jsoup.parseBodyFragment(content)
-        return doc.body().html()
+
+        //remove newline around all non block elements
```

Comment:
Right.  The nested way wouldn't change much, but the variable way is probably more readable.  I asked if you would like to take a look to get another look at the regular expressions.